django_celery_monitor/admin.py

```diff
@@ -191,6 +194,25 @@ def kill_tasks(self, request, queryset):
                 revoke(state.task_id, connection=connection,
                        terminate=True, signal='KILL')
 
+    @action(_('Retry selected tasks'))
+    def retry_tasks(self, request, queryset):
+        with current_app.default_connection() as connection:
+            for state in queryset:
+                module_name, task_name = state.name.rsplit('.', 1)
+                task = getattr(import_module(module_name), task_name, None)
+                if task:
+                    kwargs = {
+                        'args': ast.literal_eval(state.args),
+                        'kwargs': ast.literal_eval(state.kwargs),
```

Comment:
TaksState model has args and kwargs defined as TextField (https://docs.djangoproject.com/en/1.11/_modules/django/db/models/fields/#TextField), since args is tuple and kwargs is dictionary they will be converted to strings. An alternative will be to save them in database as json then we can easily decode them and not use literal_eval.